src/main/kotlin/ro/hibyte/betting/service/EventService.kt

```diff
@@ -0,0 +1,89 @@
+package ro.hibyte.betting.service
+
+import jakarta.transaction.Transactional
+import org.springframework.stereotype.Service
+import ro.hibyte.betting.dto.BetDTO
+import ro.hibyte.betting.dto.EventRequest
+import ro.hibyte.betting.dto.EventResponse
+import ro.hibyte.betting.dto.UserProfileDTO
+import ro.hibyte.betting.entity.Event
+import ro.hibyte.betting.entity.UserProfile
+import ro.hibyte.betting.mapper.BetMapper
+import ro.hibyte.betting.mapper.EventMapper
+import ro.hibyte.betting.repository.EventRepository
+import ro.hibyte.betting.repository.UserProfileRepository
+import java.lang.RuntimeException
+import java.sql.Timestamp
+import java.util.stream.Collectors
+
+@Service
+class EventService(
+    private val eventRepository: EventRepository,
+    private val eventMapper: EventMapper,
+    private val betTypeService: BetTypeService,
+    private val betMapper: BetMapper,
+    private val betService: BetService,
+    private val userProfileService: UserProfileService
+) {
+    fun addEvent(eventRequest: EventRequest) {
+        val event: Event = eventMapper.mapEventRequestToEvent(eventRequest, betTypeService)
+        eventRepository.save(event)
+    }
+
+    @Transactional
+    fun editEvent(eventId: Long, updatedEvent: EventRequest) {
+        val existingEvent = eventRepository.findById(eventId)
+        if (existingEvent.isPresent) {
```

Comment:
Here I would get rid of the `if` statement and throw the runtime exception in an `orElseThrow`:

```kotlin
val existingEvent = eventRepository.findById(eventId).orElseThrow(/*...*/)
```